### PR TITLE
chore: Bugtracker: Address github library deprecations

### PR DESCRIPTION
**Pull request by kingthorin**
- Update help to mention tokens along with passwords.
- Update UI and config classes to use "Token" for variables etc where applicable.
- Update CHANGELOG adding change note.
- Update Messages.properties key/value pairs as applicable
- Removed a few unused key/value pairs. (As a clean code activity.)
- Update build file to use the updated library.

Fixes zaproxy/zaproxy#6893

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubParam.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
     private static final String ALL_CONFIGS_KEY = GITHUB_BASE_KEY + ".configs.config";
 
     private static final String CONFIG_NAME_KEY = "name";
-    private static final String CONFIG_PASSWORD_KEY = "password";
+    private static final String CONFIG_TOKEN_KEY = "password";
```
Comment (kingthorin) [inline addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubParam.java:41]: I left the config key named as "password", this isn't using versioned config and I wasn't sure it was worth changing.

**Comment by kingthorin** [comment]
Tested with a saved session against a dummy repo, issues were successfully raised as expected.

File: addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java [hunk 2]
```diff
@@ -24,29 +24,44 @@
...
 
     public String getName() {
         return username;
     }
 
+    /**
+     * @deprecated v4 use {@link #getToken()} instead
+     * @return The password to connect.
+     */
+    @Deprecated
```
Comment (thc202) [inline addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java:52]: Not worth, the add-on is not using SemVer.

File: addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubParam.java (hunk 1 above)
Comment (thc202) [reply addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubParam.java:41]: Better change.

File: addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java (hunk 2 above)
Comment (kingthorin) [reply addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java:52]: Shall we switch it to semver? `4.0.0`?

Comment (thc202) [reply addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java:52]: I don't think it's necessary at this point, I'd expect this to be used more with the UI than programmatically (e.g. `ExtensionBugTracker` does not expose an API to raise issues).

**Comment by kingthorin** [comment]
Tweaked.

File: addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java (hunk 2 above)
Comment (thc202) [reply addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java]: I was suggesting not deprecate at all, just rename it.

Comment (kingthorin) [reply addOns/bugtracker/src/main/java/org/zaproxy/zap/extension/bugtracker/BugTrackerGithubConfigParams.java]: Oh okay, can do that too.

**Comment by kingthorin** [comment]
Addressed feedback.

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**